obspy/core/trace.py

```diff
@@ -2779,6 +3241,207 @@ def _data_sanity_checks(value):
                "arrays are allowed for initialization.") % str(value.shape)
         raise ValueError(msg)
 
+class FrequencyDomainTrace(BaseTrace):
+
+    def ifft(self):
+        tr = Trace()
+        tr.stats = Stats()
+        tr.stats.starttime = self.stats.starttime
+        tr.data = fftpack.irfft(self.data)
+        return tr
+
+    def plot(self):
+        N = self.stats.npts
+        T = self.stats.delta
+        x = np.linspace(0.0, N*T, N)
+        yf = self.data
+        xf = np.linspace(0.0, 1.0/(2.0*T), N/2)
+        fig, ax = plt.subplots()
+        ax.plot(xf, 2.0/N * np.abs(yf[:N/2]))
+        plt.xlabel('Freq (Hz)')
+        plt.ylabel('|Y(freq)|')
+        plt.show()
+
+    def differentiation(self, order_of_differention = 1):
+        self.data = fftpack.diff(self.data, order = order_of_differention)
+        return self
+
+    def integration(self, order_of_integration=1):
+        order_of_integration = (-1)*order_of_integration
+        self.data = fftpack.diff(self.data, order = order_of_integration)
+        return self
+
+    @property
+    def phase(self):
+        return np.angle(self.data)
+
+    @property
+    def amplitude(self):
+        return np.absolute(self.data)
+    
+    def cross_correlation(self, tr2):
+
+        if self.stats.starttime != tr2.stats.starttime:
+            msg = "Traces do not have the same starttime"
+            raise Exception(msg)
+       
+        elif "FALSE" == np.array_equal(self.frequencies(), tr2.frequencies()):
+            msg = "Traces do not have the same frequencies"
+            raise Exception(msg)
+
+        else: 
+            data1 = self.data
+            data2 = tr2.data 
+            data2_conj = np.conj(data2)
+            data_multiplied = np.multiply(data1, data2_conj)
+            corr = fftpack.ifft(data_multiplied)
+       
+            plt.plot(corr)        
+            plt.xlabel('offset between two waves')
+            plt.ylabel('correlation')
+            plt.show()
+
+            return (corr)        
+
+    def frequencies(self):
+        n = self.data.size
+        timestep = self.stats.delta
+        freq = fftpack.fftfreq(n, d=timestep)       
```

Comment:
You likely want to use `rfftfreq` but this is also very awkward to the previously mentioned issue.